datafusion/core/src/physical_plan/joins/hash_join.rs

```diff
@@ -1097,6 +1098,49 @@ pub fn equal_rows(
     err.unwrap_or(Ok(res))
 }
 
+fn eq_dyn_null(
```

Comment:
A null element can occur in an `Int32Array` as well. Maybe we should insert an option to arrow kernel, in arrow-rs.

Reply:
I think we can open an issue on arrow-rs for this functionality.

Reply:
https://github.com/apache/arrow-rs/issues/4438